plaso/parsers/winreg.py

```diff
@@ -212,10 +243,35 @@ def ParseFileObject(self, parser_mediator, file_object, **kwargs):
     if not root_key:
       return
 
-    try:
-      self._ParseRecurseKeys(parser_mediator, root_key)
-    except IOError as exception:
-      parser_mediator.ProduceExtractionError('{0:s}'.format(exception))
-
+    find_specs = parser_mediator.knowledge_base.GetValue(
+        artifacts_filter_file.ARTIFACTS_FILTER_FILE)
+
+    if (find_specs and
```

Comment:
flatten this if else by changing the order in which things are checked

Reply:
refactored this a bit